### PR TITLE
fix: broken github link on generated documentation pages

### LINKED ISSUE #11919: Broken link to "Edit on Github" in ipython documentation
The API documentation on readthedocs contains a link "Edit on github" in the header. Clicking on the link brings us to a 404 github page:

For example, 

Documentation page:  https://ipython.readthedocs.io/en/latest/api/generated/IPython.conftest.html

Github link: https://github.com/ipython/ipython/blob/master/docs/source/api/generated/IPython.conftest.rst

### PR DESCRIPTION
**Pull request by cocobennett**
Fix https://github.com/ipython/ipython/issues/11919

Currently trying to use the fix listed on https://github.com/sphinx-doc/sphinx/issues/2386

### CODE REVIEW

File: docs/source/_templates/breadcrumbs.html [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      {%- if prev %}
+        <a href="{{ prev.link|e }}" class="btn btn-neutral float-left" title="{{ prev.title|striptags|e }}" accesskey="p"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> {{ _('Previous') }}</a>
+      {%- endif %}
+      {%- if next %}
+        <a href="{{ next.link|e }}" class="btn btn-neutral float-right" title="{{ next.title|striptags|e }}" accesskey="n">{{ _('Next') }} <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
+      {%- endif %}
+  </div>
+  {%- endif %}
+  <hr/>
+</div>
```
Comment (MrMino) [inline docs/source/_templates/breadcrumbs.html:83]: Nitpick: could you please add an EOF newline?

**Review by MrMino — COMMENTED** [review]
Hmm... this is taken from https://github.com/readthedocs/sphinx_rtd_theme/blob/master/sphinx_rtd_theme/breadcrumbs.html, is it?

Looks like this is already inside `sphinx-rtd-theme` - why does it have to be added separately?

**Comment by cocobennett** [comment]
@MrMino Thanks, you're right. I just need to figure out how to make sure to only show the link if the url is valid

**Comment by cocobennett** [comment]
Unfortunately, I have no idea how to verify or test the documentation works here in the UI or locally, so I'm closing for now. :(

**Comment by MrMino** [comment]
Don't hesitate to ask if something is unclear :slightly_smiling_face:. You can generate the docs locally by issuing the following commands from the repository root:

```
pip install -r docs/requirements.txt pytest nose testpath
cd docs/
make html
```

Afterwards you'll find the index page of the docs under `docs/build/html/index.html`.

**Comment by cocobennett** [comment]
After being able to verify, it seems like this code resolved the issue.
Before:
image

After:
image

The after isn't pretty, but it's better than a 404!

**Comment by cocobennett** [comment]
@MrMino My last commit removes the link for all pages. It looks like the link can't be removed for only generated pages until https://github.com/readthedocs/sphinx_rtd_theme/issues/252 is fixed

**Comment by cocobennett** [comment]
Hey there, following up. Does anyone have a recommendation here?

**Comment by MrMino** [comment]
@cocobennett Sorry, got distracted and forgot to respond. Could you please summarize what is going on on the `sphinx-rtd-theme` side? The info in these issues is reliant on previous knowledge about the internals, and I don't have much of it.

**Comment by Carreau** [comment]
Oh, look like you added PNG as part of b8ade5cf639f9042411bf47fe86ebf1c6318e239, I've amended your last commit.

I think we can merge this and see how it's behaving on readthedocs and iterate.

Before merging and just to make sure, it looks like your first commit is referencing a different email than the others (potentially work email), and another have a different name, I want to make sure you are not non intentionally revealing private informations, and that this is ok. 

let us know if you need help amending the authoring informations.

**Comment by cocobennett** [comment]
> @cocobennett Sorry, got distracted and forgot to respond. Could you please summarize what is going on on the `sphinx-rtd-theme` side? The info in these issues is reliant on previous knowledge about the internals of RTD/theme, and I don't have much of it.

It looks like the issue of telling the difference between a generated page that will lead to a 404 vs an actual page has to be fixed first within `sphinx-rtd-theme` before anything can be done here to fix, or at least that's what I've gathered

**Comment by cocobennett** [comment]
> Oh, look like you added PNG as part of [b8ade5c](https://github.com/ipython/ipython/commit/b8ade5cf639f9042411bf47fe86ebf1c6318e239), I've amended your last commit.
> 
> I think we can merge this and see how it's behaving on readthedocs and iterate.
> 
> Before merging and just to make sure, it looks like your first commit is referencing a different email than the others (potentially work email), and another have a different name, I want to make sure you are not non intentionally revealing private informations, and that this is ok.
> 
> let us know if you need help amending the authoring informations.

Yes, thanks for asking! That was from a different account before I re-configured my local setup. I am okay with it staying in. Thanks for the PNG fix and for asking :)

**Comment by cocobennett** [comment]
> > @cocobennett Sorry, got distracted and forgot to respond. Could you please summarize what is going on on the `sphinx-rtd-theme` side? The info in these issues is reliant on previous knowledge about the internals of RTD/theme, and I don't have much of it.
> 
> It looks like the issue of telling the difference between a generated page that will lead to a 404 vs an actual page has to be fixed first within `sphinx-rtd-theme` before anything can be done here to fix, or at least that's what I've gathered

Well, it looks like it was fixed in the theme, so I don't know why I can't get it to tell the difference https://github.com/readthedocs/sphinx_rtd_theme/pull/393/files

**Comment by cocobennett** [comment]
Trying suggestion here https://github.com/sphinx-doc/sphinx/issues/2386#issuecomment-478403897

I tested it locally, but seems to act differently when generated in the PR, so trying both

**Comment by Carreau** [comment]
> I tested it locally, but seems to act differently when generated in the PR, so trying both

Let's try to merge and see how it affects the doc; I know that readthedocs sometime use a different version of the theme. So that may be why.

If you convert it back to non-draft I'm happy to merge this.

**Comment by Carreau** [comment]
Thanks ! Let's see how it renders on readthedocs in a couple of hours.

**Comment by MrMino** [comment]
I don't think this fixed the problem, unfortunately. The links that #11919 mentions are still visible & broken.

**Pull request merged**